zcash_client_sqlite/src/error.rs

```diff
@@ -118,17 +122,22 @@ pub enum SqliteClientError {
     /// A note selection query contained an invalid constant or was otherwise not supported.
     NoteFilterInvalid(NoteFilter),
 
-    /// The proposal cannot be constructed until transactions with previously reserved
-    /// ephemeral address outputs have been mined. The parameters are the account UUID and
-    /// the index that could not safely be reserved.
-    #[cfg(feature = "transparent-inputs")]
-    ReachedGapLimit(AccountUuid, u32),
-
-    /// An ephemeral address would be reused. The parameters are the address in string
-    /// form, and the txid of the earliest transaction in which it is known to have been
-    /// used.
+    /// An address cannot be reserved, or a proposal cannot be constructed until a transaction
+    /// containing outputs belonging to a previously reserved address has been mined. The error
+    /// contains the index that could not safely be reserved.
     #[cfg(feature = "transparent-inputs")]
-    EphemeralAddressReuse(String, TxId),
+    ReachedGapLimit(KeyScope, u32),
```

Comment:
How can you interpret the index without the account? To me, the index only makes sense in the context of an account, which is why the `AccountUuid` was originally included. Why is it hard to include?

Reply:
The account identifier is explicitly provided by the caller in every case where this error might be generated. The reason that the account UUID is hard to include is that this is generated in many circumstances where the account UUID is not known, because it's a few layers deep in having used the internal account ID for the lookup.